### ISSUE TITLE
outlines with gemma4 output wrong answer

### ISSUE THREAD

**Issue by chaowenguo** [outlines-dev/outlines#2034]
### Describe the issue as clearly as possible:

I use bitsandbytes to create q4 version of gemma4 and run in kaggle T4 x 2
```python3
!python3 -m pip install -U transformers bitsandbytes outlines
import huggingface_hub, transformers, torch, outlines, builtins
huggingface_hub.snapshot_download(repo_id='exact-railcar/gemma', local_dir='/tmp/kaggle')
processor = transformers.Gemma4Processor.from_pretrained('/tmp/kaggle')
model = transformers.Gemma4ForConditionalGeneration.from_pretrained('/tmp/kaggle', device_map='auto', dtype=torch.bfloat16)
messages = [
    {"role": "user", "content": "What is 1 + 1"},
]

# Process input
inputs = processor.apply_chat_template(
    messages,
    tokenize=True,
    return_dict=True,
    return_tensors="pt",
    add_generation_prompt=True,
    enable_thinking=True
).to(model.device)
input_len = inputs["input_ids"].shape[-1]

# Generate output
outputs = model.generate(**inputs, max_new_tokens=1024)
response = processor.decode(outputs[0][input_len:], skip_special_tokens=False)

# Parse output
processor.parse_response(response, prefix=inputs["input_ids"])
```
the output is:
```
{'role': 'assistant',
 'thinking': 'The user is asking for the sum of 1 and 1.\nBasic arithmetic. $1 + 1 = 2$.',
 'content': '1 + 1 = 2'}
```
which is correct

Now I use outline to run
```python3
model = outlines.from_transformers(transformers.Gemma4ForConditionalGeneration.from_pretrained('/tmp/kaggle', device_map='auto', dtype=torch.bfloat16), transformers.Gemma4Processor.from_pretrained('/tmp/kaggle'))
model(outlines.inputs.Chat([{"role": "user", "content": "What is 1 + 1"}]), output_type=builtins.int, temperature=0.1, max_new_tokens=1024)
```
the output is a list of 1, that is '1111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111'

### Steps/code to reproduce the bug:

```python
see above
```

### Expected result:

```shell
just 2
```

### Error message:

```shell

```

### Outlines/Python version information:

Version information
outlines-1.3.3
python3.12

### Context for the issue:

it make everything not working

**Comment by rupeshpoojary9** [comment]
Looked into this. Integer output in outlines is backed by an unbounded regex like -?\d+, so there is no natural bounded stop state the way there would be for a fixed length pattern. Once the model is inside an accepting state for that grammar, valid continuations include another digit or stopping, and stopping is supposed to happen when the model itself chooses an EOS equivalent token.

My guess is the logits processor is still masking out the EOS token while the FSM is in an accepting-but-continuable state, so the model is never actually allowed to end generation on its own, it can only keep emitting more digits until max_new_tokens is hit. That would explain the repeated 1s exactly, once the model is in a state where both digit and stop are valid the constrained decoding should allow the model to pick either, but if only digit continuation tokens are left unmasked, the model has no way to terminate early.

If that is right, the fix is in the logits processor for the integer (and likely float) Guide, once the FSM is in an accepting state, EOS or the model's actual end token needs to be added to the allowed set at each step, not just literal digit continuations. Would want to confirm this against a smaller, faster model first before touching Gemma4 specifically, to rule out a tokenizer quirk versus a general int Guide bug.

Mind if I take this one? I will start by reproducing with a small local model and inspecting exactly which tokens are allowed once the FSM reaches its accepting state for int output.
